contracts/clearing/BrokerVault.sol

```diff
@@ -43,38 +59,37 @@ contract BrokerVault is IVault2, ISettle, Ownable2Step, ReentrancyGuard {
 		broker = broker_;
 	}
 
-	function setAuthorizer(IAuthorizeV2 newAuthorizer) external onlyOwner {
-		if (address(newAuthorizer) == address(0)) {
-			revert InvalidAddress();
-		}
-
-		authorizer = newAuthorizer;
-		emit AuthorizerChanged(newAuthorizer);
-	}
-
 	// ---------- Write functions ----------
 
 	function deposit(address token, uint256 amount, address to) external payable {
-		require(msg.value == 0, 'IncorrectValue');
-		require(token != address(0), 'InvalidAddress');
+		if (msg.value == 0) {
+			revert InvalidAmount(msg.value);
+		}
+		if (token == address(0)) {
+			revert InvalidAddress();
+		}
```

Comment:
Please, add support of Native asset (address(0)), see the LiteVault implementation in the broker-contracts.
(Basically, expect a `value` equals to `amount` provided in the tx, if `token == address(0)`

Reply:
4c067cb